src/game/Player.cpp

```diff
@@ -15841,8 +15841,9 @@ void Player::SaveToDB()
         _SaveStats();
 
     // save pet (hunter pet level and experience and all type pets health/mana).
-    if (Pet* pet = GetPet())
-        pet->SavePetToDB(PET_SAVE_AS_CURRENT);
+    if (getClass() != CLASS_PRIEST)
```

Comment:
Wouldn't it be easier to place this check inside SavePetToDB itself (do it as a check before loading any data)?